pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -301,3 +302,26 @@ func (a *acrProvider) parseACRLoginServerFromImage(image string) string {
 
 	return ""
 }
+
+// shouldUseCache decide whether to use cache according to image and cache content
+func shouldUseCache(image string, d credentialprovider.DockerConfig) bool {
+	if len(d) > 2 {
+		// for service principal(len=5), always use cache
```

Comment:
how is len=5 related to len > 2?

Reply:
for managed identity case, `len(d) = 2`, that's a tick, let me add one more comment

Reply:
added comment